Library/Formula/libguestfs.rb

```diff
@@ -0,0 +1,207 @@
+  option "with-php", "Build with PHP bindings"
+
+  # Download the precompiled appliance unless explicitly told not to.
+  option "without-fixed-appliance", "Skip downloading the fixed appliance (not recommended)"
+
+  # The two required gnulib patches have been reported to gnulib mailing list, but with little effect so far.
+  patch do
+    # Add an implementation of open_memstream for BSD/Mac.
+    # Using Eric Blake's proposal originally published here: https://lists.gnu.org/archive/html/bug-gnulib/2010-04/msg00379.html
+    # and mentioned again here: http://lists.gnu.org/archive/html/bug-gnulib/2015-02/msg00083.html
+    url "https://gist.githubusercontent.com/shulima/93138eb342fe94273edd/raw/c75eac3a7f536dca526f52cd8cb5c0d6ce8beecc/gnulib-open_memstream.patch"
+    sha256 "d62f539def7300e4155bf2447b3c22049938a279957a4a97964d2d04440b58ce"
+  end
+  patch do
+    # Add a program_name equivalent for Mac.
+    # http://lists.gnu.org/archive/html/bug-gnulib/2015-02/msg00078.html
+    url "https://gist.githubusercontent.com/shulima/d851f8f35526db5e2fe9/raw/f80f6a73ec102bbdea2394d9bd3482b400853f2c/gnulib-program_name.patch"
+    sha256 "d17d1962b98a3418a335915de8a2da219e4598d42c24555bbbc5b0c1177dd38c"
+  end
+
+  # Since we can't build an appliance, the recommended way is to download a fixed one.
+  resource "fixed_appliance" do
+    url "http://libguestfs.org/download/binaries/appliance/appliance-1.28.1.tar.xz"
+    sha256 "d6aa214ba05666ca7b8494b2f1814f2f291e569a9476e837e4b9d061ff92cfa2"
+  end
+
+  def install
+    # configure doesn't detect ncurses correctly
+    ENV["LIBTINFO_CFLAGS"] = "-I/usr/include/ncurses"
+    ENV["LIBTINFO_LIBS"] = "-lncurses"
+
+    ENV["FUSE_CFLAGS"] = "-I/usr/local/include/osxfuse/fuse -D_FILE_OFFSET_BITS=64"
+    ENV["FUSE_LIBS"] = "-losxfuse"
+
+    args = [
+      "--disable-probes",
+      "--disable-appliance",
+      "--disable-daemon",
+      # Not supporting OCaml bindings due to ocamlfind (required) not being packaged in homebrew.
+      "--disable-ocaml",
+    ]
+
+    args << "--without-libvirt" if build.without? "libvirt"
+    args << "--disable-php"  if build.without? "php"
+    args << "--disable-perl" if build.without? "perl"
+
+    if build.with? "python"
+      ENV.prepend_path "PKG_CONFIG_PATH", `python-config --prefix`.chomp + "/lib/pkgconfig"
+      args << "--with-python-installdir=#{lib}/python2.7/site-packages"
+    else
+      args << "--disable-python"
+    end
+
+    if build.with? "ruby"
+      # Force ruby bindings to install locally
+      ruby_libdir = "#{lib}/ruby/site_ruby/#{RbConfig::CONFIG["ruby_version"]}"
+      ruby_archdir = "#{ruby_libdir}/#{RbConfig::CONFIG["sitearch"]}"
+      inreplace "ruby/Makefile.am", /\$\(RUBY_LIBDIR\)/, ruby_libdir
+      inreplace "ruby/Makefile.am", /\$\(RUBY_ARCHDIR\)/, ruby_archdir
+    else
+      args << "--disable-ruby"
+    end
+
+    if build.with? :java
+      args << "--with-java="+`#{Language::Java.java_home_cmd}`.chomp
```

Comment:
Probably interpolate this string rather than using `+`. @xu-cheng is there a better way to do this?